CHANGELOG.md

```diff
@@ -4,12 +4,17 @@
 
 ### New features
 
+-   Added arguments `band_prob`, `eval_points`, `rvs`, and `random_state` to `plot_ecdf` ([2316](https://github.com/arviz-devs/arviz/pull/2316))
+-   Added rcParam `plot.band_prob` ([2316](https://github.com/arviz-devs/arviz/pull/2316))
```

Comment:
Okay, so to clarify, is the idea that we replace `stats.hdi_prob` and the new `plot.band_prob` with a new `stats,ci_prob`? If so, is there a procedure for deprecating rcparams so that users can get an informative warning if they try to set `hdi_prob`?

Reply:
I don't think we have ever changed an rcParam key, only a couple values. We should definitely add a deprecation warning and keep both around for a bit.

A proposal could be to have ci_prob behave like current hdi_prob and hdi_prob now instead takes an extra value: `ci_prob` (new default). If the value for hdi_prob is different than that then raise a FutureWarning. I think we can achieve that with a custom validation function relatively easily.

Reply:
Here's what I did, roughly patterned after matplotlibs's own handing of deprecated rcparams: https://github.com/arviz-devs/arviz/pull/2316/commits/f6cff760dfe2850305792ea214bdeedfae1b10b1

Effectively, the `hdi_prob` rcparam is now an alias for `ci_prob`, which has a default. Anytime someone sets or gets `hdi_prob`, a deprecation warning is raised. The implementation is flexible enough to support in the future more complicated deprecations. At the moment this raises plenty of deprecation warnings in our tests/docs builds, since `hdi_prob` is regularly accessed. But before changing any of those things, It'd be nice to get feedback on this approach.

Reply:
This is great, much better than my proposal. The only comment is the emitted warnings should be `FutureWarning` (user facing) instead of `DeprecationWarning` (downstream dev facing). I don't really know how or why but this is Python convention and by default users don't even see `DeprecationWarning` unless explicitly activated. Ref https://docs.python.org/3/library/warnings.html#warning-categories

Reply:
Ah, good to know, thanks! Done.